### PR TITLE
Closes #2506: Removal of harvested DOI

### LINKED ISSUE #2506: Superuser should be able to delete metadata record fetched from Datacite API
* Endpoint should be available to Superuser only, so we should use token.
* It should be possible to select one or more datasets using DOI (or a list of DOIs).

### PR DESCRIPTION
**Pull request by diasf**
Support for removing harvested DOI. Refactored common part with OAI harvester.

Issue: #2506

### CODE REVIEW

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/harvest/client/DataciteDOIHarvester.java [hunk 1]
```diff
@@ -42,7 +46,7 @@ public Class<DataciteHarvesterParams> getParamsClass() {
 
     @Override
     public HarvesterResult harvest(DataverseRequest dataverseRequest, HarvestingClient harvestingClient, Logger hdLogger, DataciteHarvesterParams params) throws ImportException {
```
Comment (madryk) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/harvest/client/DataciteDOIHarvester.java:43]: Can we add test cases for this method?

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/harvest/client/DataciteDOIHarvester.java [hunk 2]
```diff
@@ -28,6 +29,9 @@ public class DataciteDOIHarvester implements Harvester<DataciteHarvesterParams>
     @Inject
     private DataciteDatasetMapper dataciteDatasetMapper;
 
+    @Inject
+    private DatasetDao datasetDao;
```
Comment (madryk) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/harvest/client/DataciteDOIHarvester.java]: Do we use this service in this class?

**Review by madryk — CHANGES_REQUESTED** [review]

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/harvest/client/DataciteDOIHarvester.java (hunk 1 above)
Comment (diasf) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/harvest/client/DataciteDOIHarvester.java:43]: Yep, added some unit tests.

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/harvest/client/DataciteDOIHarvester.java (hunk 2 above)
Comment (diasf) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/harvest/client/DataciteDOIHarvester.java]: Removed.

**Review by madryk — APPROVED** [review]

**Pull request merged**